### PR TITLE
Initial cleanup of core shell script

### PR DESCRIPTION
**Pull request by pythoninthegrass**
**NOTE**
This is a WIP meant to flag what's being done behind the scenes. Currently untested so it'd be a good idea to not approve the merge until you've ran it on your end. I'll be doing the same locally and update the PR when I've tested it.

@moraroy Please leave comments on individual TODOs / proposed changes with yea/nay and I'll honor your requests.

- use $PATH bash shebang
- capture exit code from pipes
- locally removed `.shellcheckrc`  and hardcoded `# shellcheck disable=SC____` to pinpoint language server errors
- environment variables to get actual user calling script ($HOME can be `/home/root` if called via `sudo`)
- ~~parameterize working directory~~ technically already existed, just 1000+ lines down — moved to top of the script and replaced hardcoded filepaths
- added a bunch of inline TODOs
- add error handling for killing steam by process id (pid)
- removed unnecessary newlines
- fixed indents on various code blocks
- tidied nested python script per pep8 (e.g., newlines between functions)
- set script as executable (`chmod +x`)

### CODE REVIEW

File: NonSteamLaunchers.sh [hunk 1]
```diff
@@ -1871,7 +1720,7 @@ if [[ $options == *"Origin"* ]]; then
     pkill Origin.exe
 
     # Download version.dll file
-    if [ ! -f "~/Downloads/NonSteamLaunchersInstallation/version.dll" ]; then
+    if [ ! -f "$HOME/Downloads/NonSteamLaunchersInstallation/version.dll" ]; then
         echo "Downloading version.dll file"
         wget https://github.com/p0358/Fuck_off_EA_App/releases/download/v2/version.dll -O ~/Downloads/NonSteamLaunchersInstallation/version.dll
```
Comment (pythoninthegrass) [inline NonSteamLaunchers.sh:1725]: Good name for a repo lmao

**Comment by pythoninthegrass** [comment]
For context re: `killall steam` patch:

**Before**
Screen Shot 2023-08-12 at 1 07 40 PM
Screen Shot 2023-08-12 at 1 09 15 PM
**After**
Screen Shot 2023-08-12 at 1 14 01 PM


Should probably clarify, own a Steam Deck, but in the process of migrating to the ROG Ally running a community fork of SteamOS / HoloISO, [ChimeraOS](https://chimeraos.org/).

**Comment by moraroy** [comment]
Wait wait wait this is interesting actually @pythoninthegrass Is there a way I can message you on discord or something I need to show you something haha

**Comment by pythoninthegrass** [comment]
Sure thing. My handle is `exographicskip`

File: NonSteamLaunchers.sh (hunk 1 above)
Comment (moraroy) [reply NonSteamLaunchers.sh]: wait what do you mean??? ahaha

Comment (pythoninthegrass) [reply NonSteamLaunchers.sh]: > Fuck_off_EA_App

Whatever works 👌

**Pull request merged**